py4j-java/src/main/java/py4j/ClientServer.java

```diff
@@ -160,4 +160,85 @@ public Object getPythonServerEntryPoint(Class[] interfacesToImplement) {
 		return proxy;
 	}
 
+	/**
+	 * Helper class to make it easier and self-documentating how a
+	 * {@link ClientServer} is constructed.
+	 */
+	public static class ClientServerBuilder {
+		private int fJavaPort;
+		private InetAddress fJavaAddress;
+		private int fPythonPort;
+		private InetAddress fPythonAddress;
+		private int fConnectTimeout;
+		private int fReadTimeout;
+		private ServerSocketFactory fServerSocketFactory;
+		private SocketFactory fSocketFactory;
+		private Object fEntryPoint;
+
+		public ClientServerBuilder() {
+			this(null);
+		}
+
+		public ClientServerBuilder(Object entryPoint) {
+			fJavaPort = GatewayServer.DEFAULT_PORT;
+			fJavaAddress = GatewayServer.defaultAddress();
+			fPythonPort = GatewayServer.DEFAULT_PYTHON_PORT;
+			fPythonAddress = GatewayServer.defaultAddress();
+			fConnectTimeout = GatewayServer.DEFAULT_CONNECT_TIMEOUT;
+			fReadTimeout = GatewayServer.DEFAULT_READ_TIMEOUT;
+			fServerSocketFactory = ServerSocketFactory.getDefault();
+			fSocketFactory = SocketFactory.getDefault();
+			fEntryPoint = entryPoint;
+		}
+
+		public ClientServer build() {
```

Comment:
Is it possible to add a small unit test demonstrating (and testing) the use of the builder?